bot/exts/moderation/voice_gate.py

```diff
@@ -37,208 +34,159 @@
-        except ResponseCodeError as e:
-            if e.status == 404:
-                embed = discord.Embed(
-                    title="Not found",
-                    description=(
-                        "We were unable to find user data for you. "
-                        "Please try again shortly, "
-                        "if this problem persists please contact the server staff through Modmail."
-                    ),
-                    color=Colour.red()
+            data = await self.bot.api_client.get(
+                f"bot/users/{interaction.user.id}/metricity_data",
+                raise_for_status=True
+            )
+        except ResponseCodeError as err:
+            if err.response.status == 404:
+                await interaction.response.send_message((
+                    "We were unable to find user data for you. "
+                    "Please try again shortly. "
+                    "If this problem persists, please contact the server staff through ModMail."),
+                    ephemeral=True,
+                    delete_after=GateConf.delete_after_delay,
                 )
-                log.info(f"Unable to find Metricity data about {ctx.author} ({ctx.author.id})")
+                log.info(f"Unable to find Metricity data about {interaction.user} ({interaction.user.id})")
             else:
-                embed = discord.Embed(
-                    title="Unexpected response",
-                    description=(
-                        "We encountered an error while attempting to find data for your user. "
-                        "Please try again and let us know if the problem persists."
-                    ),
-                    color=Colour.red()
+                await interaction.response.send_message((
+                    "We encountered an error while attempting to find data for your user. "
+                    "Please try again and let us know if the problem persists."),
+                    ephemeral=True,
+                    delete_after=GateConf.delete_after_delay,
                 )
-                log.warning(f"Got response code {e.status} while trying to get {ctx.author.id} Metricity data.")
-            try:
-                await ctx.author.send(embed=embed)
-            except discord.Forbidden:
-                log.info("Could not send user DM. Sending in voice-verify channel and scheduling delete.")
-                await ctx.send(embed=embed)
+                log.warning(f"Got response code {err.status} while trying to get {interaction.user.id} Metricity data.")
+            return
 
+        if interaction.user.get_role(Roles.voice_verified):
+            await interaction.response.send_message((
+                "You have already verified! "
+                "If you have received this message in error, "
+                "please send a message to the ModMail bot."),
+                ephemeral=True,
+                delete_after=GateConf.delete_after_delay,
```

Comment:
I think the button's callback is way too loaded with different responsibilities, it could help define the different "sub-boundaries" and checks better by factoring these out into separate private methods of the view.

That way it'll become easier to read & understand, and potentially refactor.

Reply:
Fair point, I'll see how I can chop it up.

Reply:
After looking this over multiple times, I just don't see the benefit of splitting it up.  The methods I would be creating would mostly be one line returns.  It feels like it ends up more.... I guess Java like, with gets that don't really benefit the code.  

I do see where it could help readability by being able to just look at the general class structure and see the method names, but at the same time, splitting it up leads to a couple complications.  For instance, for the try block, I considered making a `_get_metricity_data()` method.  But then would I have it handle the exceptions and the messages those send?  If I did, then it'd be a method that'd be doing more than its namesake and if I didn't, I'd be making a method that's adding yet another call on top of already just `send`ing the message.  

I kind of think splitting it up would hurt readability a bit, or just add method call chains that don't really need to be there.